systers_portal/meetup/views.py

```diff
@@ -12,3 +12,17 @@ class MeetupLocationAboutView(MeetupLocationMixin, TemplateView):
 
     def get_meetup_location(self):
         return get_object_or_404(MeetupLocation, slug=self.kwargs['slug'])
+
+
+class MeetupView(MeetupLocationMixin, TemplateView):
```

Comment:
In that case we need to change the `get_meetup_location()` - don't do an extra fetch, simply return `self.object`. 

Regarding the snippet you pasted here:
1. For the arg `meetup_location` you can pass the meetup location object.
2. The 404 is generated I assume because `get_queryset` only returns a Meetup object and you basically loose the MeetupLocation object. There is actually no need to override this method, simply in the `get_context_data()` do a correct fetch and pass a MeetupLocation to the query.

Is that clear?